src/parser/instruction.rs

```diff
@@ -242,6 +246,211 @@ mod tests {
+                    MemoryReference {
+                        name: "dest".to_owned(),
+                        index: 0
+                    },
+                    MemoryReference {
+                        name: "ro".to_owned(),
+                        index: 0
+                    },
+                    TernaryOperand::LiteralInteger(0)
+                )
+            }),
+            Instruction::TernaryLogic(TernaryLogic {
+                operator: TernaryOperator::LessThan,
+                operands: (
+                    MemoryReference {
+                        name: "dest".to_owned(),
+                        index: 0
+                    },
+                    MemoryReference {
+                        name: "ro".to_owned(),
+                        index: 1
+                    },
+                    TernaryOperand::LiteralInteger(-1)
+                )
+            }),
+            Instruction::TernaryLogic(TernaryLogic {
+                operator: TernaryOperator::LessThanOrEqual,
+                operands: (
+                    MemoryReference {
+                        name: "dest".to_owned(),
+                        index: 0
+                    },
+                    MemoryReference {
+                        name: "ro".to_owned(),
+                        index: 0
+                    },
+                    TernaryOperand::LiteralReal(1.2)
+                )
+            }),
+            Instruction::TernaryLogic(TernaryLogic {
+                operator: TernaryOperator::GreaterThan,
+                operands: (
+                    MemoryReference {
+                        name: "dest".to_owned(),
+                        index: 0
+                    },
+                    MemoryReference {
+                        name: "ro".to_owned(),
+                        index: 2
+                    },
+                    TernaryOperand::LiteralReal(0.000001)
+                )
+            }),
+            Instruction::TernaryLogic(TernaryLogic {
+                operator: TernaryOperator::GreaterThanOrEqual,
+                operands: (
+                    MemoryReference {
+                        name: "dest".to_owned(),
+                        index: 0
+                    },
+                    MemoryReference {
+                        name: "ro".to_owned(),
+                        index: 0
+                    },
+                    TernaryOperand::MemoryReference(MemoryReference {
+                        name: "x".to_owned(),
+                        index: 0
+                    }),
+                )
+            })
+        ]
+    );
+
+    #[test]
+    fn test_ternary_logic_error() {
+        [
+            "EQ ro 1 1",
+            "LT 1 1 1",
+            "LE 1 x ro",
+            "GT 1 ro x",
+            "GE dest 0.3 4",
+        ]
+        .iter()
+        .for_each(|input| {
+            let tokens = lex(input).unwrap();
+            assert!(parse_instructions(&tokens).is_err())
```

Comment:
Might be worth adding an assertion message with the input string so it's clear which one fails